Lib/configparser.py

```diff
@@ -687,7 +688,7 @@ def read(self, filenames, encoding=None):
 
         Return list of successfully read files.
         """
-        if isinstance(filenames, str):
+        if isinstance(filenames, (str, bytes, os.PathLike)):
```

Comment:
I think I see what you mean, I do think both things need to be addressed but it does make sense to split them out.